src/ions/store/store.jsx

```diff
@@ -10,6 +10,14 @@ const useStore = create(set => ({
 			})
 		);
 	},
+	selectedLesson: 1,
+	setSelectedLesson: number => {
+		set(
+			produce(state => {
+				state.selectedLesson = number;
+			})
+		);
+	},
```

Comment:
for this usecase, using immer.js is pretty overengineered. 

`setSelectedLesson: number => {
		set(
			state => ({
				selectedLesson: number;
			})
		);
	},`